opentelemetry-sdk/src/testing/metrics/in_memory_exporter.rs

```diff
@@ -285,6 +285,9 @@ impl TemporalitySelector for InMemoryMetricsExporter {
 #[async_trait]
 impl PushMetricsExporter for InMemoryMetricsExporter {
     async fn export(&self, metrics: &mut ResourceMetrics) -> Result<()> {
+        if metrics.scope_metrics.is_empty() || metrics.scope_metrics[0].metrics.is_empty() {
```

Comment:
this is unexpected, right? if nothing to be exported, the sdk should not call exporters? Or this is to signal to exporters that export is trigged, but have no metric to export.
(unrelated to this PR I guess...)

Reply:
Yes, I did notice the export triggered with empty metric data. Let me check it further if it is related to the PR. Don't merge till I resolve this comment :)

Reply:
Ok, have removed this check. Seems it was caused by some thread conflict caused by running multiple tests concurrently in my local setup. The CI runs them sequentially so it shouldn't be an issue.